assets/js/job-submission.js

```diff
@@ -3,8 +3,63 @@ jQuery(document).ready(function($) {
 		$(this).closest( '.job-manager-uploaded-file' ).remove();
 		return false;
 	});
-	$( document.body ).on( 'submit', '.job-manager-form:not(.prevent-spinner-behavior)', function() {
+	$( document.body ).on( 'submit', '.job-manager-form:not(.prevent-spinner-behavior)', function( event ) {
+		// Validate the category field if present
+		var jobCategory = $( '#job_category' );
+		if ( jobCategory.length && !jobCategory.val() ) {
+			event.preventDefault();
+			$(this).find( 'input[type=submit]' ).blur();
+
+			var jobCategoryInput = $( '.select2-search__field' )[0];
+			jobCategoryInput.setCustomValidity( 'Category is required' );
+			jobCategoryInput.reportValidity();
+			
+			return;
+		}
+
+		// Validate the description field if present
+		if ( typeof tinymce !== undefined || tinymce.get( 'job_description' ) != null ) {
+			var jobDescription = tinymce.get('job_description').getContent();
+			var editorTextArea = $( '#job_description' );
```

Comment:
Technically this could also be made optional, but I'm not as concerned about this field. Seems pretty critical to the CPT. 

If we wanted to, it would be passing a `required` class in `editor_class` in [this template's](https://github.com/Automattic/wp-job-manager/blob/332601223930d0a4e26b327700f754c705421e50/templates/form-fields/wp-editor-field.php) editor arguments.